src/layout-persistence/dotnet/src/MorganStanley.ComposeUI.LayoutPersistence/Abstractions/ILayoutSerializer.cs

```diff
@@ -0,0 +1,22 @@
+﻿/*
+ * Morgan Stanley makes this available to you under the Apache License,
+ * Version 2.0 (the "License"). You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership. Unless required by applicable law or agreed
+ * to in writing, software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+namespace MorganStanley.ComposeUI.LayoutPersistence.Abstractions
+{
+    public interface ILayoutSerializer
+    {
+        string Serialize<T>(T layoutObject);
```

Comment:
I think these should be asynchronous. Eventually we probably need to serialize state which can only be retrieved from web apps in an asynchronous manner.